src/test/java/hudson/plugins/git/extensions/impl/PruneStaleTagTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+package hudson.plugins.git.extensions.impl;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.File;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.commons.io.FileUtils;
+import org.jenkinsci.plugins.gitclient.GitClient;
+import org.jenkinsci.plugins.gitclient.TestJGitAPIImpl;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import hudson.EnvVars;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.plugins.git.GitSCM;
+import hudson.util.LogTaskListener;
+import nl.jqno.equalsverifier.EqualsVerifier;
+
+public class PruneStaleTagTest {
+
+    @Rule
+    public TemporaryFolder fileRule = new TemporaryFolder();
+
+    @Test
+    public void equalsContract() {
+        EqualsVerifier.forClass(PruneStaleTag.class)
+                .usingGetClass()
+                .verify();
+    }
+
+    @Test
+    public void testFetch() throws Exception {
+        PruneStaleTag extension = new PruneStaleTag();
+        
+        TaskListener listener = new LogTaskListener(Logger.getGlobal(), Level.OFF);
+        Run<?, ?> run = mock(Run.class);
+        when(run.getEnvironment(listener)).thenReturn(new EnvVars());
+        
+        File gitRoot = fileRule.newFolder("remote");
+        FileUtils.touch(new File(gitRoot, "test"));
+        String gitURL = gitRoot.toURI().toString();
+        String tagName = "tag";
+
+        // create a remote repository with one tag
+        GitClient remoteClient = new TestJGitAPIImpl(gitRoot, listener);
```

Comment:
Does not test the command line git implementation.

Needs a test of command line git implementation.